oauth2_lib/fastapi.py

```diff
@@ -305,64 +305,64 @@ async def _opa_decision(
+        requestMethod = request.method if hasattr(request, "method") else "GET"
+        opa_input = {
+            "input": {
+                **(opa_kwargs or {}),
+                **user_info,
+                "resource": request.url.path,
+                "method": requestMethod,
+                "arguments": {"path": request.path_params, "query": {**request.query_params}, "json": json},
             }
+        }
 
-            logger.debug("Posting input json to Policy agent", input=opa_input)
-
-            try:
-                result = await async_request.post(opa_url, json=opa_input)
-            except (NetworkError, TypeError):
-                raise HTTPException(status_code=HTTPStatus.SERVICE_UNAVAILABLE, detail="Policy agent is unavailable")
-
-            data = OPAResult.parse_obj(result.json())
-
-            if not data.result and auto_error:
-                logger.debug(
-                    "User is not allowed to access the resource",
-                    decision_id=data.decision_id,
-                    resource=request.url.path,
-                    method=requestMethod,
-                    user_info=user_info,
-                    input=opa_input,
-                    url=request.url,
-                )
-                raise HTTPException(
-                    status_code=HTTPStatus.FORBIDDEN,
-                    detail=f"User is not allowed to access resource: {request.url.path} Decision was taken with id: {data.decision_id}",
-                )
-            if data.result:
-                logger.debug(
-                    "User is authorized to access the resource",
-                    decision_id=data.decision_id,
-                    resource=request.url.path,
-                    method=requestMethod,
-                    user_info=user_info,
-                    input=opa_input,
-                )
-
-            return data.result
+        logger.debug("Posting input json to Policy agent", input=opa_input)
 
-        return None
+        try:
+            result = await async_request.post(opa_url, json=opa_input)
+        except (NetworkError, TypeError):
+            raise HTTPException(status_code=HTTPStatus.SERVICE_UNAVAILABLE, detail="Policy agent is unavailable")
+
+        data = OPAResult.parse_obj(result.json())
+
+        if not data.result and auto_error:
+            logger.debug(
+                "User is not allowed to access the resource",
+                decision_id=data.decision_id,
+                resource=request.url.path,
+                method=requestMethod,
+                user_info=user_info,
+                input=opa_input,
+                url=request.url,
+            )
+            raise HTTPException(
+                status_code=HTTPStatus.FORBIDDEN,
+                detail=f"User is not allowed to access resource: {request.url.path} Decision was taken with id: {data.decision_id}",
+            )
+        if data.result:
```

Comment:
Is this check needed? In other words: could we have no auth_error but also at the same time no data.result?

Reply:
That is possible if any of the applications call this function with `auto_error=False`

Reply:
So basically the possible branches are 
```
if data.result:
  # User is authorized
  return True

# Not authorized
if auto_error:
  raise ...
  
return False
```

I will rewrite it so both `opa_decision()` and `opa_graphql_decision()` can reuse this decision making.
Also I'll make `opa_graphql_decision()` function actually check `auto_error` because now it always behaves as if `auto_error=False`

Reply:
Refactored it, could you have another look?